core/src/main/scala/org/apache/spark/ui/jobs/JobPage.scala

```diff
@@ -335,9 +335,12 @@ private[ui] class JobPage(parent: JobsTab, store: AppStatusStore) extends WebUIP
 
     content ++= makeTimeline(activeStages ++ completedStages ++ failedStages,
       store.executorList(false), appStartTime)
-
-    content ++= UIUtils.showDagVizForJob(
-      jobId, store.operationGraphForJob(jobId))
+    try {
+      content ++= UIUtils.showDagVizForJob(
+        jobId, store.operationGraphForJob(jobId))
+    } catch {
+      case e => None
+    }
```

Comment:
Same here. We should avoid the situation when the exception is thrown. Catching the exception and doing nothing just hides the problems.